subprojects/platform-jvm/src/main/java/org/gradle/platform/BuildPlatform.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.platform;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Information about the machine host Gradle is running on.
+ *
+ * @since 7.6
+ */
+@Incubating
+public interface BuildPlatform {
```

Comment:
This API needs more work, I am not convinced about using enums for example.

Reply:
I am wondering about this too. The current HEAD still uses enums. What was the resolution of this discussion?